src/views/TheChatbox.vue

```diff
@@ -326,28 +332,34 @@ export default {
         console.error("Error sending POST request:", error);
       }
     },
+    resetVotingState(masterName) {
+      if (this.votingStates[masterName]) {
+        this.votingStates[masterName] = {};
+      }
+    },
 
-    async vote(isUpvote, masterName) {
+    async vote(isUpvote, masterName,answerIndex) {
       // Find the master's message based on the master's name
-      console.log(masterName);
-      const message = this.messages.find((msg) => msg.author === masterName);
+      const message = this.messages.find((msg) => msg.author === masterName && msg.answerIndex === answerIndex);
 
       if (!message) {
         console.error(`Master ${masterName}'s answer not found.`);
         return;
       }
       // Check if the master is already voted
-      if (
-        this.votingStates[masterName] &&
-        this.votingStates[masterName][isUpvote ? "upvote" : "downvote"]
-      ) {
+     // Check if the answer is already voted
+    const isAnswerUpvoted = this.votingStates[`${masterName}-${answerIndex}`]?.upvote;
+    const isAnswerDownvoted = this.votingStates[`${masterName}-${answerIndex}`]?.downvote; 
+
+    if ((isUpvote && isAnswerUpvoted) || (!isUpvote && isAnswerDownvoted)) {
         alert(
           `You have already voted ${
             isUpvote ? "👍" : "👎"
           } for ${masterName}'s answer.`
         );
         return;
       }
```

Comment:
Why is this code needed when you have already disabled the buttons if the answer is present in voting states object?